components/Tabs/src/TabBarView/MDCTabBarView.m

```diff
@@ -352,11 +359,29 @@ - (void)observeValueForKeyPath:(NSString *)keyPath
 - (void)layoutSubviews {
   [super layoutSubviews];
 
-  CGFloat availableWidth = CGRectGetWidth(self.bounds);
+  CGRect availableBounds = self.bounds;
+  if (@available(iOS 11.0, *)) {
+    availableBounds = UIEdgeInsetsInsetRect(availableBounds, self.safeAreaInsets);
+  }
+  CGFloat availableWidth = CGRectGetWidth(availableBounds);
   CGFloat requiredWidth = [self justifiedWidth];
   BOOL canBeJustified = availableWidth >= requiredWidth;
-  self.containerView.distribution = canBeJustified ? UIStackViewDistributionFillEqually
-                                                   : UIStackViewDistributionFillProportionally;
+  if (canBeJustified) {
+    self.containerView.distribution = UIStackViewDistributionFillEqually;
+    if (@available(iOS 11.0, *)) {
+      [NSLayoutConstraint deactivateConstraints:self.widthConstraints];
+      [NSLayoutConstraint activateConstraints:self.justifiedSafeAreaInsetConstraints];
+    }
+  } else {
+    self.containerView.distribution = UIStackViewDistributionFillProportionally;
+    // Inset the safe area insets if it is a scrollable view.
+    if (@available(iOS 11.0, *)) {
+      [NSLayoutConstraint deactivateConstraints:self.justifiedSafeAreaInsetConstraints];
+      self.contentInset = UIEdgeInsetsMake(self.contentInset.top, self.safeAreaInsets.left,
```

Comment:
I believe `adjustedContentInset` already includes the `safeAreaInsets` if `contentInsetAdjustmentBehavior` is `.automatic`. Did you not find that to be the case?

Reply:
Some further reading:
https://medium.com/@wailord/the-particulars-of-the-safe-area-and-contentinsetadjustmentbehavior-in-ios-11-9b842018eeaa

Reply:
Able to use `safeAreaLayoutGuide` and `contentLayoutGuide` for the appropriate layout styles. Setting the `contentInsetsAdjustmentBehavior` to `.always` seems to be the correct way to get a UIScrollView to position its content inside the safe area.